### PR TITLE
Hygiene fixes in doxygen documentation

### PR DESCRIPTION
**Pull request by aggarw13**
This PR makes fixes to doxygen documentation to make a clear distinction between **corePKCS11 library** and the **PKCS #11 standard** as well as add mechanism for keeping the configuration-macros up-to-date.

### CODE REVIEW

File: docs/doxygen/pages.dox [hunk 1]
```diff
@@ -21,10 +21,10 @@ This PKCS #11 library implements a subset of the PKCS #11 API required to establ
 
 /**
 @page pkcs11_design Design
-@section PKCS11_Wrapper PKCS #11 Wrapper Dependencies
-Currently, the PKCS #11 wrapper library has a dependency on:
-- The C standard library stdint
-- PKCS #11.
+@section PKCS11_Wrapper Dependencies of corePKCS11 Wrapper for PKCS #11 standard
+Currently, the corePKCS11 library provides a wrapper header around the PKCS #11 standard. It has depdnencies on:
```
Comment (paulbartell) [inline docs/doxygen/pages.dox:25]:
```suggestion
Currently, the corePKCS11 library provides a wrapper header around the PKCS #11 standard. It has dependencies on:
```

File: docs/doxygen/pages.dox [hunk 2]
```diff
@@ -21,10 +21,10 @@ This PKCS #11 library implements a subset of the PKCS #11 API required to establ
...
+- The C standard library `stdint`
```
Comment (paulbartell) [inline docs/doxygen/pages.dox:26]:
```suggestion
- `stdint` from the C standard library
```

File: docs/doxygen/pages.dox [hunk 3]
```diff
@@ -21,10 +21,10 @@ This PKCS #11 library implements a subset of the PKCS #11 API required to establ
...
+- PKCS #11 standard.
```
Comment (paulbartell) [inline docs/doxygen/pages.dox:27]: Perhaps this could be a link to the spec from OASIS to make it more clear?

Comment (aggarw13) [reply docs/doxygen/pages.dox:27]: Added link to OASIS

File: docs/doxygen/pages.dox [hunk 4]
```diff
@@ -7,8 +7,8 @@
 
 <span style="float:right;margin-right:4em"> &mdash; <i>Official documentation of PKCS #11 from [oasis](http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html)</i></span><br>
 
-This PKCS #11 library implements a subset of the PKCS #11 API required to establish a secure connection to AWS IoT:
-- Verifying the contents of a message.
+This corePKCS11 library implements a subset of the PKCS #11 API required to establish a secure connection to AWS IoT:
+- Verifying signature on the contents of a message.
```
Comment (archigup) [inline docs/doxygen/pages.dox:11]:
```suggestion
- Verifying the signature of the contents of a message.
```

File: docs/doxygen/pages.dox [hunk 5]
```diff
@@ -21,10 +21,10 @@ This PKCS #11 library implements a subset of the PKCS #11 API required to establ
 
 /**
 @page pkcs11_design Design
-@section PKCS11_Wrapper PKCS #11 Wrapper Dependencies
-Currently, the PKCS #11 wrapper library has a dependency on:
-- The C standard library stdint
-- PKCS #11.
+@section PKCS11_Wrapper Dependencies of corePKCS11 Wrapper for PKCS #11 standard
```
Comment (archigup) [inline docs/doxygen/pages.dox:24]:
```suggestion
@section PKCS11_Wrapper Dependencies of the corePKCS11 Wrapper for the PKCS #11 standard
```

File: docs/doxygen/pages.dox [hunk 6]
```diff
@@ -83,11 +83,13 @@ digraph pkcs11_software_implementation_dependencies
 }
 @enddot
 
-@section PKCS11_utilities PKCS #11 Utilities Dependencies
+@section PKCS11_utilities corePKCS1111 Utilities Dependencies
```
Comment (archigup) [inline docs/doxygen/pages.dox:86]:
```suggestion
@section PKCS11_utilities corePKCS11 Utilities Dependencies
```

File: docs/doxygen/pages.dox [hunk 7]
```diff
@@ -83,11 +83,13 @@ digraph pkcs11_software_implementation_dependencies
...
 
-The PKI utils module is a forked version of the PKI utilities provided by mbed TLS. They are used to convert from
-Currently, the module has the following dependencies:
-- The standard C library
+The PKI utils module is a forked version of the PKI utilities provided by Mbed TLS. They provide helper utilities to convert format of ECDSA P-256 signatures.
```
Comment (archigup) [inline docs/doxygen/pages.dox:88]:
```suggestion
The PKI utils module is a forked version of the PKI utilities provided by Mbed TLS. They provide helper utilities to convert the format of ECDSA P-256 signatures.
```

File: docs/doxygen/pages.dox [hunk 8]
```diff
@@ -83,11 +83,13 @@ digraph pkcs11_software_implementation_dependencies
...
+- DER to format required for PKCS #11
+- PKCS #11 to ASN.1 format
```
Comment (archigup) [inline docs/doxygen/pages.dox:90]:
```suggestion
The conversions provider are:
- DER format to PKCS #11 format
- PKCS #11 format to ASN.1 format
```

**Review by archigup — DISMISSED** [review]

**Review by archigup — APPROVED** [review]

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,8 @@
 # Change Log for corePKCS11 Library
 
+## v3.2.0 (August 2021)
+- [#122] Add `core_pkcs11_config_defaults.h` file for default definition of configuration macros. and make doxygen documentation fixes.
```
Comment (aggarw13) [inline CHANGELOG.md:4]: The minor version is bumped for the changes that are slated in #121

File: docs/doxygen/pages.dox [hunk 10]
```diff
@@ -21,10 +21,10 @@ This PKCS #11 library implements a subset of the PKCS #11 API required to establ
 
 /**
 @page pkcs11_design Design
-@section PKCS11_Wrapper PKCS #11 Wrapper Dependencies
-Currently, the PKCS #11 wrapper library has a dependency on:
-- The C standard library stdint
-- PKCS #11.
+@section PKCS11_Wrapper Dependencies of the corePKCS11 Wrapper for the PKCS #11 standard
```
Comment (aggarg) [inline docs/doxygen/pages.dox:24]:
```suggestion
@section PKCS11_Wrapper Dependencies of the corePKCS11 Library
```

File: docs/doxygen/pages.dox [hunk 11]
```diff
@@ -21,10 +21,10 @@ This PKCS #11 library implements a subset of the PKCS #11 API required to establ
...
+Currently, the corePKCS11 library provides a wrapper header around the [PKCS #11 standard](http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html). It has dependencies on:
+- `stdint` from the C standard library
```
Comment (aggarg) [inline docs/doxygen/pages.dox:26]:
```suggestion
- `stdint.h` from the C standard library
```

File: docs/doxygen/pages.dox [hunk 12]
```diff
@@ -48,16 +48,16 @@ digraph pkcs11_wrapper_dependencies
 }
 @enddot
 
-@section PKCS11_implementation PKCS #11 Software Implementation Dependencies
+@section PKCS11_implementation corePKCS11 Software Implementation Dependencies
 
-PKCS #11 is a flexible library, and can easily be swapped out for other implementations. The PKCS #11 library documented here
-is a software based implementation, to allow for writing libraries and code that can easily interface with Hardware Security Modules (HSM).
+The implementation that corePKCS11 provides for the PKCS #11 wrapper header, `core_pkcs11.h`, is flexible library to be swapped out for other implementations. The corePKCS11 library documented here
```
Comment (aggarg) [inline docs/doxygen/pages.dox:53]:
```suggestion
The corePKCS11 library provides one implementation of the PKCS #11 standard, and ti can easily be swapped out for other implementations. The corePKCS11 library documented here
```

File: docs/doxygen/pages.dox [hunk 13]
```diff
@@ -83,11 +83,14 @@ digraph pkcs11_software_implementation_dependencies
...
-@section PKCS11_utilities PKCS #11 Utilities Dependencies
+@section PKCS11_utilities corePKCS11 Utilities Dependencies
 
-The PKI utils module is a forked version of the PKI utilities provided by mbed TLS. They are used to convert from
-Currently, the module has the following dependencies:
-- The standard C library
+The PKI utils module is a forked version of the PKI utilities provided by Mbed TLS. They provide helper utilities to convert the format of ECDSA P-256 signatures.
+The conversions provided are:
+- DER format to PKCS #11 format
+- PKCS #11 format to ASN.1 format
```
Comment (aggarg) [inline docs/doxygen/pages.dox:91]:
```suggestion
- DER format to PKCS #11 format.
- PKCS #11 format to ASN.1 format.
```

File: source/include/core_pkcs11_config_defaults.h [hunk 14]
```diff
@@ -0,0 +1,349 @@
...
+#endif
+
+/**
+ * @brief Definitions mapping deprecated configuration macro names to their current equivalent
+ * configurations for backwards compatibility of API.
+ */
+#ifndef DOXYGEN
+    #define PKCS11_MALLOC                    pkcs11configPKCS11_MALLOC
+    #define PKCS11_FREE                      pkcs11configPKCS11_FREE
+    #define configPKCS11_DEFAULT_USER_PIN    pkcs11configPKCS11_MALLOC
```
Comment (aggarg) [inline source/include/core_pkcs11_config_defaults.h:345]:
```suggestion
    #define configPKCS11_DEFAULT_USER_PIN    pkcs11configPKCS11_DEFAULT_USER_PIN
```

File: test/system-test/test-config/core_pkcs11_config.h [hunk 15]
```diff
@@ -51,7 +51,7 @@
 #endif
 
 #ifndef LIBRARY_LOG_LEVEL
-    #define LIBRARY_LOG_LEVEL    LOG_NONE
+    #define LIBRARY_LOG_LEVEL    LOG_DEBUG
```
Comment (aggarg) [inline test/system-test/test-config/core_pkcs11_config.h:54]: Is it intentional?

File: source/include/core_pkcs11_config_defaults.h [hunk 16]
```diff
@@ -0,0 +1,349 @@
...
+
+/**
+ * @brief The PKCS #11 label for the object to be used for code verification.
+ *
+ * Used by AWS IoT Over-the-Air Update (OTA) code to verify an incoming signed image.
+ *
+ * <b>Possible values:</b> Any String smaller then pkcs11configMAX_LABEL_LENGTH.<br>
+ * <b>Default value:</b> `Code Verify Key`
+ */
+#ifndef pkcs11configLABEL_CODE_VERIFICATION_KEY
```
Comment (aggarg) [inline source/include/core_pkcs11_config_defaults.h:254]: Seems duplicated on line #228.

File: source/include/core_pkcs11_config_defaults.h [hunk 17]
```diff
@@ -0,0 +1,349 @@
...
+
+/**
+ * @file core_pkcs11_config_defaults.h
+ * @brief List of configuration macros for the corePKCS11 library along with
+ * their default values.
+ */
+
+#ifndef CORE_PKCS11_CONFIG_DEFAULTS_H_
+#define CORE_PKCS11_CONFIG_DEFAULTS_H_
+
```
Comment (aggarg) [inline source/include/core_pkcs11_config_defaults.h:34]: Add 
```
/* *INDENT-OFF* */
#ifdef __cplusplus
    extern "C" {
#endif
```

File: test/system-test/test-config/core_pkcs11_config.h (hunk 15 above)
Comment (aggarw13) [reply test/system-test/test-config/core_pkcs11_config.h]: Reverted

File: source/include/core_pkcs11_config_defaults.h (hunk 16 above)
Comment (aggarw13) [reply source/include/core_pkcs11_config_defaults.h]: Removed the duplication

File: source/include/core_pkcs11_config_defaults.h (hunk 17 above)
Comment (aggarw13) [reply source/include/core_pkcs11_config_defaults.h:34]: As this library is included in the corePKCS11 library files, each of which already have the header guards, it is not necessarily required to add it there. We also have a mechanism present in C-SDK to catch in the future if any required file is missing the header guards for the library.

File: test/system-test/test-config/core_pkcs11_config.h [hunk 18]
```diff
@@ -51,7 +51,7 @@
 #endif
 
 #ifndef LIBRARY_LOG_LEVEL
-    #define LIBRARY_LOG_LEVEL    LOG_DEBUG
+    #define LIBRARY_LOG_LEVEL    LOG_NONE
```
Comment (archigup) [inline test/system-test/test-config/core_pkcs11_config.h]: Is logging being turned off for tests?

Comment (aggarw13) [reply test/system-test/test-config/core_pkcs11_config.h]: It is enabled in CI through compile parameter though: https://github.com/FreeRTOS/corePKCS11/blob/main/.github/workflows/ci.yml#L23

File: source/include/core_pkcs11_config_defaults.h (hunk 17 above)
Comment (aggarw13) [reply source/include/core_pkcs11_config_defaults.h:34]: We discussed offline, and I added the header guards in the `config_defaults.h` file for reviewer preference.

**Review by aggarg — APPROVED** [review]

**Pull request merged**